src/react-imgix-bg.jsx

```diff
@@ -0,0 +1,122 @@
+import React from "react";
+import Measure, { withContentRect } from "react-measure";
+import constructUrl from "./constructUrl";
+import targetWidths from "./targetWidths";
+const PACKAGE_VERSION = require("../package.json").version;
+
+const noop = () => {};
+
+const findNearestWidth = actualWidth =>
+  targetWidths.reduce((currentCandidate, candidateWidth) => {
+    // <= ensures that the largest value is used
+    if (
+      Math.abs(candidateWidth - actualWidth) <=
+      Math.abs(currentCandidate - actualWidth)
+    ) {
+      return candidateWidth;
+    }
+    return currentCandidate;
+  }, Number.MAX_VALUE);
+
+const BackgroundImpl = props => {
+  const {
+    measureRef,
+    measure,
+    contentRect,
+    imgixParams = {},
+    onLoad,
+    disableLibraryParam,
+    src,
+    children,
+    className = ""
+  } = props;
+  const { w: forcedWidth, h: forcedHeight } = imgixParams;
+  const hasDOMDimensions = contentRect.bounds.top != null;
+  const htmlAttributes = props.htmlAttributes || {};
+  const dpr = imgixParams.dpr || window.devicePixelRatio || 1;
+  const ref = htmlAttributes.ref;
+  const onRef = el => {
+    measureRef(el);
+    if (typeof ref === "function") {
+      ref(el);
+    }
+  };
+
+  const { width, height } = (() => {
+    const bothWidthAndHeightPassed =
+      forcedWidth != null && forcedHeight != null;
+
+    if (bothWidthAndHeightPassed) {
+      return { width: forcedWidth, height: forcedHeight };
+    }
+
+    if (!hasDOMDimensions) {
+      return { width: undefined, height: undefined };
+    }
+    const ar = contentRect.bounds.width / contentRect.bounds.height;
+
+    const neitherWidthNorHeightPassed =
+      forcedWidth == null && forcedHeight == null;
+    if (neitherWidthNorHeightPassed) {
+      const width = findNearestWidth(contentRect.bounds.width);
+      const height = Math.ceil(width / ar);
+      return { width, height };
+    }
+    if (forcedWidth != null) {
+      const height = Math.ceil(forcedWidth / ar);
+      return { width: forcedWidth, height };
+    } else if (forcedHeight != null) {
+      const width = Math.ceil(forcedHeight * ar);
+      return { width, height: forcedHeight };
+    }
+  })();
+  const isReady = width != null && height != null;
+
+  const commonProps = {
+    ...htmlAttributes
+  };
+
+  if (!isReady) {
+    return (
+      <div
+        {...commonProps}
+        className={`react-imgix-bg-loading ${className}`}
+        ref={onRef}
+      >
+        {children}
+      </div>
+    );
+  }
+
+  const renderedSrc = (() => {
+    const srcOptions = {
+      ...imgixParams,
+      ...(disableLibraryParam ? {} : { ixlib: `react-${PACKAGE_VERSION}` }),
+      width,
+      height,
+      fit: "crop",
+      backgroundSize: "cover",
```

Comment:
Ah, isn't this accidentally adding `&backgroundSize=cover` to all of our imgix URLs?

Reply:
Whoops, don't know how this slipped in here.